### ISSUE TITLE
Cannot continue training from checkpoint

### ISSUE THREAD

**Issue by jmoreno11** [IBM/terratorch#621]
**Describe the issue**
I tried to continue fine-tuning Prithvi-v2-BurnScars on a custom data module, but I get this error when passing the checkpoint (from HuggingFace) to the trainer.

It is definitely a checkpoint issue, since I can run evaluation over the same data module.

**To Reproduce (optional, but appreciated)**
1. Create any GenericNonGeoSegmentationDataModule
2. Create a SemanticSegmentationTask (following terratorch tutorial)
3. Create a Trainer (following tutorial)
4. Try fitting trainer

**Screenshots or log output (optional)**

<img width="1354" alt="Image" src="https://github.com/user-attachments/assets/3573f293-57c6-4450-9478-05dda3f9d085" />

**Expected behavior**
Continue training.

**Deployment information**
Describe what you've deployed and how:
 - TerraTorch version: 1.0
 - Installation source: pip

**Comment by Joao-L-S-Almeida** [comment]
Hi, @jmoreno11 . Thanks, I'll try to reproduce your issue.

**Comment by jmoreno11** [comment]
Thanks! @Joao-L-S-Almeida

**Comment by Joao-L-S-Almeida** [comment]
Hi, @jmoreno11 . I'm not sure about how you have configured your experiment (feel free to share it with us, if you can), but I tried to reproduce it in the notebook https://github.com/IBM/terratorch/blob/fix/621/examples/tutorials/restore_from_checkpointing.ipynb, but that's is apparently working as expected.  How have you configured your callbacks ?

**Comment by jmoreno11** [comment]
@Joao-L-S-Almeida sorry for the late response. Here's a link to my Google Colab notebook, you should have viewer access: https://colab.research.google.com/drive/1ZMMgFpnwNU7FG_wXZZG3AmzUvNs0PG08?usp=sharing


Unfortunately, I can no longer access the notebook you shared. Any help would be truly appreciated!

**Comment by Joao-L-S-Almeida** [comment]
Hi. The link became invalid when the PR was merged. Here it is: https://github.com/IBM/terratorch/blob/main/examples/tutorials/restore_from_checkpointing.ipynb

**Comment by jmoreno11** [comment]
thanks for the quick response! I tried it but I still get the same error. I'm trying to load the checkpoint from HuggingFace, could it be that it's missing from there?

<img width="872" alt="Image" src="https://github.com/user-attachments/assets/7f6b7927-790b-4952-9c0b-e6da84085bd0" />

**Comment by Joao-L-S-Almeida** [comment]
I thought you was trying to restore from a checkpoint you created. The field `optimizer_states` in missing in these checkpoints stored on HuggingFace. Below I share the keys contained in a checkpoint I created locally:
```
In [12]: content_ = torch.load("/home/jalmeida/Projetos/terratorch/tests/all_ecos_random/version_4/checkpoints/epoch=1.ckpt", map_loc
    ...: ation=torch.device('cpu'))

In [13]: content_.keys()
Out[13]: dict_keys(['epoch', 'global_step', 'pytorch-lightning_version', 'state_dict', 'loops', 'callbacks', 'optimizer_states', 'lr_schedulers', 'hparams_name', 'hyper_parameters'])

```

**Comment by Joao-L-S-Almeida** [comment]
Next week I'll ask the folks responsible for this example in order to see whether they can share a more complete checkpoint.

**Comment by jmoreno11** [comment]
@Joao-L-S-Almeida If it's any help, I'll recreate it today-tomorrow, since I need it for my research and it's a bit time sensitive, so I can share it. I'll use the configuration available in HF, seed and everything.

**Comment by Joao-L-S-Almeida** [comment]
All right, @jmoreno11.  I'm sorry I can't help you with this file. Once you have created a checkpoint with the optimizer states, you should be able of restarting the training in the same way seen in the notebook.

**Comment by jmoreno11** [comment]
thanks for the support anyway!

**Comment by Joao-L-S-Almeida** [comment]
@jmoreno11 I'm including @blumenstiel in this conversation in order to check with him whether it's possible to send a more complete version of the Burn Scars checkpoints to HuggingFace.

**Comment by blumenstiel** [comment]
@jmoreno11, if you like, you can just re-train the model on the BurnScars dataset, it does not take long.
We provide the config here: https://huggingface.co/ibm-nasa-geospatial/Prithvi-EO-2.0-300M-BurnScars/blob/main/burn_scars_config.yaml
Here is a tutorial if you need it: https://github.com/IBM/terratorch/blob/main/examples/tutorials/PrithviEOv2/prithvi_v2_eo_300_tl_unet_burnscars.ipynb

I can have a look in the next days if we have the full checkpoint stored, but it might be that we just saved with `weights_only=True`.

**Comment by jmoreno11** [comment]
@blumenstiel I did just that, using that same configuration. @Joao-L-S-Almeida as mentioned, if you're interested, I could compatible checkpoints for the Burn Scars segmentation of Prithvi-v1 and Prithvi-v2-300, as none of their Hugging Face versions are working.

P.S. @blumenstiel I also opened an issue ([issue#3](https://github.com/IBM/terramind/issues/3)) on Terramind, since I want to do the same thing for the TiM version. It'd be great to get a hint 🙏

**Comment by Joao-L-S-Almeida** [comment]
@jmoreno11 That would be very helpful. Thank you for this effort. 
As you already defined a path to continue, I'm closing this issue. 
Feel free to open a new one for the next steps.

**Comment by blumenstiel** [comment]
Thanks @jmoreno11! I checked our servers and we only saved the weights, sorry.
